tools/src/main/java/org/apache/kafka/tools/LeaderElectionCommand.java

```diff
@@ -99,8 +99,12 @@ static void run(Duration timeoutMs, String... args) throws Exception {
             props.putAll(Utils.loadProps(commandOptions.getAdminClientConfig()));
         }
         props.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, commandOptions.getBootstrapServer());
-        props.setProperty(AdminClientConfig.DEFAULT_API_TIMEOUT_MS_CONFIG, Long.toString(timeoutMs.toMillis()));
-        props.setProperty(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, Long.toString(timeoutMs.toMillis() / 2));
+        if (!props.containsKey(AdminClientConfig.DEFAULT_API_TIMEOUT_MS_CONFIG)) {
+            props.setProperty(AdminClientConfig.DEFAULT_API_TIMEOUT_MS_CONFIG, Long.toString(timeoutMs.toMillis()));
+        }
+        if (!props.containsKey(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG)) {
+            props.setProperty(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, Long.toString(timeoutMs.toMillis() / 2));
```

Comment:
while you are over here in this part of the code can you also make another change please? This should be Int.toString since request.timeout is an int. See: https://kafka.apache.org/documentation.html#producerconfigs_request.timeout.ms

same for default.api.timeout.ms 
https://kafka.apache.org/documentation.html#adminclientconfigs_default.api.timeout.ms

Reply:
thanks for the review @divijvaidya I will do the change you proposed and add the test

Reply:
I will need to convert the timeoutMs.toMillis()  to INT as the return value is long, this comes from [here](https://github.com/apache/kafka/pull/15030/files#diff-7f2ce49cbfada7fbfff8453254f686d835710e2ee620b8905670e69ceaaa1809R66)  I will convert it in Int then like this `props.setProperty(AdminClientConfig.DEFAULT_API_TIMEOUT_MS_CONFIG, Integer.toString((int)timeoutMs.toMillis()));` agree?

Reply:
sure